crates/ruff_python_formatter/README.md

```diff
@@ -130,8 +142,73 @@ impl FormatNodeRule<ExprList> for FormatExprList {
 }
 ```
 
-Comments are categorized into `Leading`, `Trailing` and `Dangling`, you can override this in
-`place_comment`.
+A related common challenge is that we want to attach comments to tokens (think keywords and
+syntactically meaningful characters such as `:`) that have no node on their own. A slightly
+simplified version of the `while` node in our AST looks like the following:
+
+```rust
+pub struct StmtWhile {
+    pub range: TextRange,
+    pub test: Box<Expr<TextRange>>,
+    pub body: Vec<Stmt<TextRange>>,
+    pub orelse: Vec<Stmt<TextRange>>,
+}
+```
+
+That means in
+
+```python
+while True:  # Trailing cond comment
```

Comment:
i had to read the section below several times to follow (including why you were talking about the "trailing cond" comment when fixing the "leading else" comment. if i understand correctly it's because those get lumped together into the dangling_comments list. if that's correct could we write that out? though reading yet again i'm not even sure i did understand correctly. 

sorry for the back-and forth on this!

on the other hand, this pr definitely improves on the current state so we could also just land it and iterate further later

Reply:
thanks for the feedback, i made the problem of the dangling list throwing different kinds of dangling comments together more explicit